cpp/include/cudf/aggregation.hpp

```diff
@@ -259,16 +261,64 @@ std::unique_ptr<Base> make_collect_list_aggregation(
- * considered equal
+ * @param nulls_equal Flag to specify whether null entries within each list should be considered
+ *        equal.
+ * @param nans_equal Flag to specify whether NaN values in floating point column should be
+ *        considered equal.
  */
 template <typename Base = aggregation>
 std::unique_ptr<Base> make_collect_set_aggregation(null_policy null_handling = null_policy::INCLUDE,
                                                    null_equality nulls_equal = null_equality::EQUAL,
                                                    nan_equality nans_equal = nan_equality::UNEQUAL);
 
+/**
+ * @brief Factory to create a MERGE_LISTS aggregation.
+ *
+ * This aggregation is similar to `COLLECT_LIST` with the following differences:
+ *  - It requires the input values to be a lists column, and
+ *  - The value (lists) corresponding to the same key will not result in a list of lists as output
+ *    from `COLLECT_LIST`. Instead, those lists will result in a list generated by merging them
+ *    together.
+ *
+ * In practice, this aggregation is used to merge the partial results of multiple (distributed)
+ * groupby `COLLECT_LIST` aggregations into a final `COLLECT_LIST` result. Those distributed
+ * aggregations were executed on different values columns partitioned from the original values
+ * column, then their results were (vertically) concatenated before given as the values column for
+ * this aggregation.
+ *
+ * Note that this aggregation does not accept any null handling parameter as it is designed for
+ * simply merging lists of the input lists column. As such, the input lists column is required to
+ * NOT be nullable.
+ *
+ */
+template <typename Base = aggregation>
+std::unique_ptr<Base> make_merge_lists_aggregation();
+
+/**
+ * @brief Factory to create a MERGE_SETS aggregation.
+ *
+ * This aggregation is similar to `COLLECT_SET` with the following differences:
+ *  - It requires the input values to be a lists column, and
+ *  - The value (lists) corresponding to the same key will result in a list generated by merging
+ *    them together then dropping the duplicate entries.
+ *
+ * In practice, this aggregation is used to merge the partial results of multiple (distributed)
+ * groupby `COLLECT_LIST/COLLECT_SET` aggregations into a final `COLLECT_SET` result. Those
+ * distributed aggregations were executed on different values columns partitioned from the original
+ * values column, then their results were (vertically) concatenated before given as the values
+ * column for this aggregation.
+ *
+ * Note that the (distributed) partial results should be generated by `COLLECT_LIST` aggregations
+ * to avoid the overhead of calling `lists::drop_list_duplicates` before merging lists.
+ *
+ * Similar to `MERGE_LISTS`, this aggregation does not accept any null handling parameter and
+ * requires the input lists column to NOT be nullable.
```

Comment:
stale comment?

Reply:
Wrong comment instead. Thanks for pointing this out.